DiscordBot.Commands/HelpCommand.cs

```diff
@@ -0,0 +1,178 @@
+﻿using Discord;
+using Discord.WebSocket;
+using DiscordBot.Commands.Core;
+using DiscordBot.Commands.Core.CommandAttributes;
+using Microsoft.Extensions.Logging;
+
+namespace DiscordBot.Commands
+{
+    /// <summary>
+    /// Class for handling help command
+    /// </summary>
+    [Name("help")]
+    [Description("Prints help command")]
+    [Option("command", "displays command details", CommandOptionType.String, false)]
+    public class HelpCommand : Command 
+    {        
+        private readonly ILogger<HelpCommand> _logger;
+        private Lazy<IEnumerable<ICommand>> _commands;
+
+        public HelpCommand(ILogger<HelpCommand> logger, Lazy<IEnumerable<ICommand>> commands)
+        {
+            _logger = logger;
+            _commands = commands;
+        }
+
+        /// <summary>
+        /// Method for obtaining command to specific user guild permissions
+        /// </summary>
+        /// <param name="userPermissions">Discord guild permission</param>
+        /// <returns>List of command with command that this user can use</returns>
+        private List<ICommand> GetCommands(GuildPermissions userPermissions)
```

Comment:
@Niewidzialny84  I get that you hace used list to be able to filter out the commands, although you could have used IEnumerable.Where() for that. But is there a point in returning the value as editable list?